libmamba/src/core/virtual_packages.cpp

```diff
@@ -236,7 +236,52 @@ namespace mamba
             {
                 return "x86_64_v2";
             }
+#elif defined(_MSC_VER) && defined(_M_X64)
+            // We apply boolean masks to the results of the cpuid instruction
+            // to determine if the CPU supports the required instruction sets.
+            // The masks are generated by shifting 1 to the left by the bit
+            // position of the required feature.
+            //
+            // See those sources to understand bitshifts:
+            // https://docs.microsoft.com/en-us/cpp/intrinsics/cpuid-cpuidex?view=msvc-160
+            // https://en.wikipedia.org/wiki/CPUID#EAX=7,_ECX=0:_Extended_Features
+
+            int cpu_info[4];
+            int eax_value = 7;
+            int ecx_value = 0;
+            __cpuidex(cpu_info, eax_value, ecx_value);
+
+            bool avx512f = cpu_info[1] & (1 << 16);
+            bool avx512dq = cpu_info[1] & (1 << 17);
+            bool avx512cd = cpu_info[1] & (1 << 28);
+            bool avx512bw = cpu_info[1] & (1 << 30);
+            bool avx512vl = cpu_info[1] & (1 << 31);
+            if (avx512f && avx512dq && avx512cd && avx512bw && avx512vl)
+            {
+                return "x86_64_v4";
+            }
+
+            bool bmi = cpu_info[1] & (1 << 3);
+            bool avx2 = cpu_info[1] & (1 << 5);
+            bool bmi2 = cpu_info[1] & (1 << 8);
+            bool fma_ = cpu_info[1] & (1 << 12);
+            bool avx = cpu_info[1] & (1 << 28);
+            if (bmi && avx2 && bmi2 && fma_ && avx)
```

Comment:
Similar remark here: avx2 is implmeented on top of avx, so no need to check for avx.